.github/workflows/cd.yaml

```diff
@@ -0,0 +1,129 @@
+name: CD
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+
+jobs:
+  interface:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    defaults:
+      run:
+        working-directory: ./interface
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+
+      - name: Setup Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+
+      - name: Install dependencies
+        run: yarn install --nonInteractive --frozen-lockfile --prefer-offline
+
+      - name: Build
+        run: yarn build
+
+      - name: Deploy
+        run: yarn deploy
+
+      - name: Check for changes in URI.txt
+        run: |
+          git diff --name-only | grep ^URI.txt$
+          echo "changed=$?" >> $GITHUB_ENV
+
+      - name: Commit files
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add .
+          git commit -m "Add deployments" -a
+        if: env.changed == 0
+
+      - name: Push changes
+        run: |
+          git branch -c main artifacts_interface_${{ github.run_id }}
+          git push origin artifacts_interface_${{ github.run_id }}
+        if: env.changed == 0
+
+      - name: Create PR
+        run: |
+          gh pr create --title "New deployment artifacts" --body "" --base main --head artifacts_interface_${{ github.run_id }}
+        if: env.changed == 0
+
+  getImplementations:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ env.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo "matrix=$(./scripts/getImplementations.sh)" >> $GITHUB_ENV
+
+  implementations-cd:
+    runs-on: ubuntu-latest
+    needs:
+      - interface
+      - getImplementations
+    strategy:
+      matrix: ${{fromJSON(needs.getImplementations.outputs.matrix)}}
+    defaults:
+      run:
+        working-directory: ./implementations/${{ matrix.package }}
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+        working-directory: ./interface
+
+      - name: Setup Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+
+      - name: Install dependencies
+        run: yarn install --nonInteractive --frozen-lockfile --prefer-offline
+        working-directory: ./interface
```

Comment:
this won't be needed

Reply:
I think the issue might be that the build step for the interface was missing

Reply:
Maybe we need both install and build? or do we need neither?

Reply:
For now I will assume we need to install dependencies and build the interface, as you did in the CI

Reply:
This is if we go forward with this idea: https://github.com/polywrap/wrap-abi-bindgen/pull/4/files/d39c8546c28d1174bc31062bbb6ce2bcff13b264#r1245654301

Reply:
nvm, It's not super important! just something I thought of that can make CI/CD less verbose but there maybe cons of it that I haven't thought of. We can have a question issue for this and get opinions on it. I personally like building interfaces before building impls as a part of yarn script because this also helped during local development by making sure we are using the latest interface while building for implementation.